src/index.js

```diff
@@ -39,8 +31,9 @@ function formatter(values, param, type, onlyIfExists) {
   }
 
   if (value !== undefined) {
-    if (type === 'query' && !isURIEncoded(value)) {
-      return encodeURIComponent(value);
+    // Query params should always be formatted, even if they don't have a `style` serialization configured.
+    if (type === 'query') {
+      return formatStyle(value, param);
```

Comment:
we could technically do this with all params. every param type has a default style

Reply:
The defaults for the `encode(Primitive|Array|Object)()` methods right now is `return undefined`